src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -63,24 +63,32 @@ protected bool AreContainingSymbolLocalsZeroed
 
         internal void ReportAsyncParameterErrors(BindingDiagnosticBag diagnostics, Location location)
         {
-            foreach (var parameter in Parameters)
+            var parameters = !this.IsStatic ? this.GetParametersIncludingExtensionParameter() : Parameters;
```

Comment:
This could be also simplified if we checked `IsStatic` inside `GetParametersIncludingExtensionParameter`.